tests/model/lib/test_submission_process.py

```diff
@@ -245,8 +247,9 @@ def test_experiment_creation(self):
         compartments = _save_compartments(self.db_session, submission_form, study)
 
         experiments = _save_experiments(self.db_session, submission_form, study)
+        submission_form.save()
 
-        self.db_session.flush()
+        self.db_session.commit()
```

Comment:
Committing actually stores things into the database and assigns them ids visible to queries. If we use `flush`, the session is supposedly persisted, but in a transaction, which is not committed yet: https://stackoverflow.com/a/4202016

This complexity when it comes to "saving the data" is very annoying, and I plan to document it in the developer documentation.